### PR TITLE
Sorry until the end of `Retype_C.thy`

### PR DESCRIPTION
**Pull request by michaelmcinerney**

### CODE REVIEW

File: proof/crefine/RISCV64/Retype_C.thy [hunk 1]
```diff
@@ -1611,16 +1616,18 @@ proof (intro impI allI)
   apply (clarsimp simp: hrs_htd_update ptr_retyps_htd_safe_neg szo
                         kernel_data_refs_domain_eq_rotate
                         ht_rl foldr_upd_app_if [folded data_map_insert_def]
-                        rl projectKOs cvariable_array_ptr_retyps[OF szo]
+                        rl cvariable_array_ptr_retyps[OF szo]
                         zero_ranges_ptr_retyps
               simp del: endpoint_C_size)
+  (* FIXME RT: refill_buffer_relation *)
+  subgoal sorry
```
Comment (lsf37) [inline proof/crefine/RISCV64/Retype_C.thy:1623]: If you put the `FIXME` into the same line as the `sorry`, `git grep sorry` will give a nice overview of what kinds of sorries are left to do.

**Review by lsf37 — APPROVED** [review]
Nice progress!

File: proof/crefine/RISCV64/Retype_C.thy [hunk 2]
```diff
@@ -6712,8 +6760,7 @@ lemma range_cover_one:
   done
 
 lemma createObject_no_inter:
-notes blah[simp del] =  atLeastAtMost_iff atLeastatMost_subset_iff atLeastLessThan_iff
-      Int_atLeastAtMost atLeastatMost_empty_iff split_paired_Ex
+notes blah[simp del] = atLeastAtMost_simps
```
Comment (corlewis) [inline proof/crefine/RISCV64/Retype_C.thy:6763]: I've seen you make the following change elsewhere and I prefer it
```suggestion
notes [simp del] = atLeastAtMost_simps
```

File: proof/crefine/RISCV64/Retype_C.thy [hunk 3]
```diff
@@ -7258,8 +7312,7 @@ lemma range_cover_n_le':
   done
 
 lemma createNewObjects_ccorres:
-notes blah[simp del] =  atLeastAtMost_iff atLeastatMost_subset_iff atLeastLessThan_iff
-      Int_atLeastAtMost atLeastatMost_empty_iff split_paired_Ex
+notes blah[simp del] = atLeastAtMost_simps
```
Comment (corlewis) [inline proof/crefine/RISCV64/Retype_C.thy:7315]:
```suggestion
notes [simp del] = atLeastAtMost_simps
```

**Review by corlewis — APPROVED** [review]

File: proof/crefine/RISCV64/Retype_C.thy (hunk 3 above)
Comment (michaelmcinerney) [reply proof/crefine/RISCV64/Retype_C.thy:7315]: The proof for this result is currently sorried, so I'm not sure whether the `blah` name can be removed: it does currently occur in the (commented-out part of the) proof. I'll put a `FIXME RT` asking whether we can remove the `blah` name when the proof is done

**Pull request merged**